### ISSUE TITLE
LLVM build fails on OS X

### ISSUE THREAD

**Issue by pramodk** [spack/spack#2992]
On OS-X (El Capitan) I am trying : 

```
spack install -v llvm +shared_libs %clang
```
and this gives : 

```
cd /private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/lib/External 
&& /Users/kumbhar/workarena/software/sources/spack/lib/spack/env/clang/clang  -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/lib/External -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/pet/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/ppcg/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/ppcg/imath -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/lib/External/ppcg -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/isl -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/isl/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/isl/imath -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/lib/External/isl -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/tools/polly/lib/External/isl/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/JSON/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/spack-build/include -I/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/include  -fPIC -Wall -W -Wno-unused-parameter -Wwrite-strings -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wdelete-non-virtual-dtor -Werror=date-time -O3 -DNDEBUG   -fvisibility=hidden -w -std=gnu99 -o CMakeFiles/PollyISL.dir/isl/isl_version.c.o   -c /private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-YifTzq/llvm-3.9.0.src/tools/polly/lib/External/isl/isl_version.c
Undefined symbols for architecture x86_64:
  "MachProcess::GetCPUType()", referenced from:
      DNBProcessGetCPUType(int) in DNB.cpp.o
  "MachProcess::ReadMemory(unsigned long long, unsigned long, void*)", referenced from:
      DNBProcessMemoryRead(int, unsigned long long, unsigned long, void*) in DNB.cpp.o
  "MachProcess::GetPThreadT(unsigned long long)", referenced from:
      DNBGetPThreadT(int, unsigned long long) in DNB.cpp.o
  "MachProcess::SetExitInfo(char const*)", referenced from:
      DNBProcessSetExitInfo(int, char const*) in DNB.cpp.o
  "MachProcess::WriteMemory(unsigned long long, unsigned long, void const*)", referenced from:
      DNBProcessMemoryWrite(int, unsigned long long, unsigned long, void const*) in DNB.cpp.o
  "MachProcess::LookupSymbol(char const*, char const*)", referenced from:
```

haven't looked into details yet, here is [discussion thread](https://github.com/Homebrew/homebrew-core/issues/3030) on homebrew.

**Comment by eschnett** [comment]
Version 3.9.1 is out; maybe it has some bug fixes?

**Comment by pramodk** [comment]
haven't tried `3.9.1` yet but trying `trunk` gives : 

```
spack install --keep-stage -v llvm@trunk  %clang
.................
[  3%] Linking CXX shared library ../../../lib/libomptarget.dylib
cd /private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-9V99mB/trunk/spack-build/projects/openmp/libomptarget && /usr/local/Cellar/cmake/3.6.3/bin/cmake -E cmake_link_script CMakeFiles/omptarget.dir/link.txt --verbose=1
/Users/kumbhar/workarena/software/sources/spack/lib/spack/env/clang/clang++  -std=c++11  -fPIC -fvisibility-inlines-hidden -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -Werror=date-time -std=c++11 -O3 -DNDEBUG -dynamiclib -Wl,-headerpad_max_install_names  -o ../../../lib/libomptarget.dylib -install_name /private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-9V99mB/trunk/spack-build/lib/libomptarget.dylib CMakeFiles/omptarget.dir/src/omptarget.cpp.o -ldl -Wl,--version-script=/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-9V99mB/trunk/projects/openmp/libomptarget/exports
ld: unknown option: --version-script=/private/var/folders/xz/f6__x5d52vg07njslp6lndr8004_5m/T/kumbhar/spack-stage/spack-stage-9V99mB/trunk/projects/openmp/libomptarget/exports
clang-3.9: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libomptarget.dylib] Error 1
make[1]: *** [projects/openmp/libomptarget/CMakeFiles/omptarget.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
...............
```
So OS X `ld` doesn't support `--version-script` which is [used](http://lists.llvm.org/pipermail/cfe-dev/2013-September/032177.html) if `cross-compiling`.

**Comment by alalazo** [comment]
The problem is reported to be solved in newer versions in #3683. Please reopen if this is still an issue.
